contracts/HybridCustody.cdc

```diff
@@ -0,0 +1,529 @@
+    also maintains a set of accounts that it "owns", meaning it has a capability to the full ChildAccount resource and
+    would then also be able to manage the child account's links as it sees fit.
+*/
+pub contract HybridCustody {
+
+    pub let ChildStoragePath: StoragePath
+    pub let ChildPublicPath: PublicPath
+    pub let ChildPrivatePath: PrivatePath
+
+    pub let ManagerStoragePath: StoragePath
+    pub let ManagerPublicPath: PublicPath
+    pub let ManagerPrivatePath: PrivatePath
+
+    pub let LinkedAccountPrivatePath: PrivatePath
+    pub let BorrowableAccountPrivatePath: PrivatePath
+
+    // TODO: Events!
+
+    // An interface which gets shared to a Manager when it is given full ownership of an account.
+    pub resource interface Account {
+        pub fun getAddress(): Address
+        pub fun isChildOf(_ addr: Address): Bool
+        pub fun getParentsAddresses(): [Address]
+        pub fun borrowAccount(): &AuthAccount?
+    }
+
+    // A ChildAccount shares the BorrowableAccount capability to itelf with ProxyAccount resources
+    pub resource interface BorrowableAccount {
+        access(contract) fun borrowAccount(): &AuthAccount
+    }
+
+    // Public methods anyone can call on a child account
+    pub resource interface ChildAccountPublic {
+        pub fun getParentsAddresses(): [Address]
+        pub fun getParentStatuses(): {Address: Bool}
+        pub fun getRedeemedStatus(addr: Address): Bool?
+
+        access(contract) fun setRedeemed(_ addr: Address)
+    }
+
+    // Accessible to the owner of the ChildAccount.
+    pub resource interface ChildAccountPrivate {
+        pub fun removeParent(parent: Address): Bool
+        pub fun publishToParent(
+            parentAddress: Address,
+            factory: Capability<&CapabilityFactory.Manager{CapabilityFactory.Getter}>,
+            filter: Capability<&{CapabilityFilter.Filter}>
+        )
+        pub fun giveOwnership(to: Address)
+        pub fun relinquishOwnership()
+    }
+
+    // Public methods exposed on a proxy account resource. ChildAccountPublic will share
+    // some methods here, but isn't necessarily the same
+    pub resource interface AccountPublic {
+        pub fun getPublicCapability(path: PublicPath, type: Type): Capability?
+        pub fun getAddress(): Address
+    }
+
+    // Methods only accessible to the designated parent of a ProxyAccount
+    pub resource interface AccountPrivate {
+        pub fun getCapability(path: CapabilityPath, type: Type): Capability?
+        pub fun getPublicCapability(path: PublicPath, type: Type): Capability?
+
+        access(contract) fun redeemedCallback(_ addr: Address)
+    }
+
+    // Entry point for a parent to borrow its child account and obtain capabilities or
+    // perform other actions on the child account
+    pub resource interface ManagerPrivate {
+        pub fun borrowAccount(id: UInt64): &{AccountPrivate, AccountPublic}?
```

Comment:
Would also be helpful to borrow an account by Address. Also wondering if we should add a method to borrow owned accounts in this interface.

Reply:
I think yes on both counts, here

Reply:
Added these methods, but added a comment about what we should be able to obtain from an owned account. We might need a CapabilityFactory instance there as well for the owner to maintain what you can borrow straight from there (there is no proxy in that scenario.)